planner/core/find_best_task.go

```diff
@@ -811,7 +811,10 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty, planCounter
 				p: dual,
 			}, cntPlan, nil
 		}
-		canConvertPointGet := len(path.Ranges) > 0 && path.StoreType == kv.TiKV && ds.isPointGetConvertableSchema()
+		canConvertPointGet := len(path.Ranges) > 0 && path.StoreType == kv.TiKV && ds.isPointGetConvertableSchema() &&
+			// to avoid the over-optimized risk, do not generate PointGet for plan cache, for example,
+			// `pk>=$a and pk<=$b` can be optimized to a PointGet when `$a==$b`, but it can cause wrong results when `$a!=$b`.
+			!ds.ctx.GetSessionVars().StmtCtx.UseCache
```

Comment:
Here we cannot generate `PointGet` if prepared-cache-plan is enabled. Will that be a problem in OLTP workload? For example, as for `prepare stmt1 from "insert into t2 select * from t1 where a=?"`, if the sql will be executed frequently, we may hope to not only generate `PointGet` plan but also cache the plan.